examples/daal/cpp_sycl/source/datasource/row_merged_to_sycl_homogen.cpp

```diff
@@ -0,0 +1,144 @@
+  covAlg.input.set(covariance::data, table);
+  covAlg.parameter.outputMatrixType = covariance::correlationMatrix;
+  covAlg.compute();
+
+  return covAlg.getResult()->get(covariance::correlation);
+}
+
+/* Fill the buffer with pseudo random numbers generated with MinStd engine */
+void generateData(float *dataBlock, size_t nRows, size_t nCols) {
+  for (size_t i = 0; i < nRows; i++) {
+    constexpr float genMax = 2147483647.0f;
+    uint32_t genState = 7777 + i * i;
+    genState = generateMinStd(genState);
+    genState = generateMinStd(genState);
+    for (size_t j = 0; j < nCols; j++) {
+      dataBlock[i * nCols + j] = (float)genState / genMax;
+      genState = generateMinStd(genState);
+    }
+  }
+}
+
+int main(int argc, char *argv[]) {
+  constexpr size_t nCols = 10;
+  constexpr size_t nRows = 10000;
+
+  for (const auto &deviceDescriptor : getListOfDevices()) {
+    const auto &device = deviceDescriptor.second;
+    const auto &deviceName = deviceDescriptor.first;
+    std::cout << "Running on " << deviceName << std::endl << std::endl;
+
+    /* Crate SYCL* queue with desired device */
+    cl::sycl::queue queue{device};
+
+    /* Set the queue to default execution context */
+    Environment::getInstance()->setDefaultExecutionContext(
+        SyclExecutionContext{queue});
+
+    SharedPtr<RowMergedNumericTable> mergedTable =
+        RowMergedNumericTable::create();
+
+    std::vector<float *> memChunks;
+    bool allocatedAllBlocks = true;
+
+    for (size_t i = 0; i < 3; i++) {
+      /* Allocate memory on host to store input data */
+      float *dataBlock = (float *)malloc(sizeof(float) * nRows * nCols);
+      if (!dataBlock) {
+        std::cout << "Allocation failed" << std::endl;
+        allocatedAllBlocks = false;
+        for (size_t i = 0; i < memChunks.size(); i++) {
+          free(memChunks[i]);
+        }
+        break;
+      }
+      memChunks.push_back(dataBlock);
+
+      /* Fill allocated memory block with generated numbers */
+      generateData(dataBlock, nRows, nCols);
+
+      /* Create numeric table from shared memory */
+      NumericTablePtr dataTable =
+          HomogenNumericTable<float>::create(dataBlock, nCols, nRows);
+
+      /* Add to row merged table */
+      mergedTable->addNumericTable(dataTable);
+    }
+    if (!allocatedAllBlocks) {
+      continue;
+    }
+
+    /* Convert row merged table to sycl homogen one */
+    Status st;
+    NumericTablePtr tablePtr = convertToSyclHomogen<float>(*mergedTable, st);
+    if (!st.ok()) {
+      std::cout << "Failed to convert row merged table to SYCL homogen one"
+                << std::endl;
+      return -1;
+    }
+
+    /* Compute correlation matrix of generated dataset */
+    NumericTablePtr covariance = computeCorrelationMatrix(tablePtr);
+
+    /* Print the results */
+    printNumericTable(covariance, "Covariance matrix:");
+
+    /* Free data blocks*/
+    for (size_t i = 0; i < memChunks.size(); i++) {
+      free(memChunks[i]);
+    }
```

Comment:
Move to separate function `deallocateMemoryChunks`

Reply:
Done